src/com/xilinx/rapidwright/edif/EDIFHierNet.java

```diff
@@ -201,19 +201,29 @@ public List<EDIFHierPortInst> getLeafHierPortInsts() {
      * aliases.
      */
     public List<EDIFHierPortInst> getLeafHierPortInsts(boolean includeSourcePins) {
-        return getLeafHierPortInsts(includeSourcePins, new HashSet<>());
+        return getLeafHierPortInsts(includeSourcePins, true);
+    }
+
+    public List<EDIFHierPortInst> getLeafHierPortInsts(boolean includeSourcePins, boolean includeSinkPins) {
+        return getLeafHierPortInsts(includeSourcePins, includeSinkPins, new HashSet<>());
+    }
+
+    public List<EDIFHierPortInst> getLeafHierPortInsts(boolean includeSourcePins, Set<EDIFHierNet> visited) {
+        return getLeafHierPortInsts(includeSourcePins, true, visited);
     }
 
     /**
      * Gets all connected leaf port instances on this hierarchical net and its aliases.
-     * @param includeSourcePins A flag to include source pins in the result.  Setting this to false
-     * only returns the sinks.
+     * @param includeSourcePins A flag to include source pins in the result.
+     * @param includeSinkPins A flag to include sink pins in the result.
      * @param visited An initial set of EDIFHierNet-s that have already been visited and will not
      * be visited again. Pre-populating this set can be useful for blocking traversal.
      * @return The list of all leaf cell port instances connected to this hierarchical net and its
      * aliases.
      */
-    public List<EDIFHierPortInst> getLeafHierPortInsts(boolean includeSourcePins, Set<EDIFHierNet> visited) {
+    public List<EDIFHierPortInst> getLeafHierPortInsts(boolean includeSourcePins, boolean includeSinkPins, Set<EDIFHierNet> visited) {
+        assert(includeSourcePins || includeSinkPins);
```

Comment:
I'm not sure I agree that both flags being `false` indicates an error state.  If I was looking for just `INOUT` ports, I could see that being useful.  Perhaps we could clarify in the Javadoc that when both flags are false, it will return any `INOUT` ports?

Reply:
Hmm. According to my testing, `EDIFPort{Inst,HierInst}.isInput()` and `EDIFPort{Inst,HierInst}.isOutput()` will return false for `INOUT`-s. This looks like the only way to tell an inout. Does this match your understanding?

For this method, it looks like inout-s (on blackbox cells, as I don't think it makes sense for a FPGA primitive to have a inout that isn't connected to a top-level port) and thus are never picked up. This should probably change.

Reply:
Ok, let's change the comment to declare that this method will never return `INOUT` ports/portInsts and that setting both flags to `false` will always return an empty list.